src/components/projectSelect/index.tsx

```diff
@@ -4,35 +4,57 @@ import Cookies from "js-cookie";
 import { useHistory } from "react-router-dom";
 import SearchableDropdown from "components/SearchableDropdown";
 import { CURRENT_PROJECT } from "constants/cookies";
-import { getCommitsRoute } from "constants/routes";
 import {
   GetProjectsQuery,
   GetProjectsQueryVariables,
+  GetViewableProjectRefsQuery,
+  GetViewableProjectRefsQueryVariables,
 } from "gql/generated/types";
-import { GET_PROJECTS } from "gql/queries";
+import { GET_PROJECTS, GET_VIEWABLE_PROJECTS } from "gql/queries";
 import { ProjectOptionGroup } from "./ProjectOptionGroup";
 
 interface ProjectSelectProps {
   selectedProjectIdentifier: string;
+  isProjectSettingsPage?: boolean;
```

Comment:
This components implementation is still tied to closely to the projectSettings page.

Reply:
the fact that there is a special prop called `isProjectSettingsPage` makes this component less generic then it should be. I think just renaming this to the condition it toggles will make it more generic. 
`canClickOnRepoGroup` or something like that.

Reply:
This naming feels less intuitive to me. 
If I was a new engineer reading this code, it would make more sense to me that we are using different queries to fetch the projects based on `isProjectSettingsPage` or `!isProjectSettingsPage`. It would be less apparent why we are doing that based on wether or not we want the headers to be clickable. Clickable headers are only one part of what distinguishes the project page dropdown component. The main difference is which projects are being fetched. 

I'll rename it for the ProjectOptionGroup component though.